### PR TITLE
Closes #1875 duration and interval time difference types update

### LINKED ISSUE #1875: Feature Request:  Enhance `compute_duration()` to handle Interval and Duration
### Feature Idea

`lubridate::time_length(x, unit = "second")` handles [Interval](https://lubridate.tidyverse.org/reference/Interval-class.html) and [Duration](https://lubridate.tidyverse.org/reference/Duration-class.html) differently when unit is months/years

> When x is an [Interval](https://lubridate.tidyverse.org/reference/Interval-class.html) object and unit are years or months, time_length() takes into account the fact that all months and years don't have the same number of days.
> When x is a [Duration](https://lubridate.tidyverse.org/reference/Duration-class.html), [Period](https://lubridate.tidyverse.org/reference/Period-class.html) or [difftime()](https://rdrr.io/r/base/difftime.html) object, length in months or years is based on their most common lengths in seconds (see [timespan()](https://lubridate.tidyverse.org/reference/timespan.html)).

Example of difference between intervals and durations:
```r
> int <- interval(ymd("2000-02-01"), ymd("2000-03-01"))
> time_length(int, "months")
[1] 1
> time_length(as.duration(int), "months")
[1] 0.9527721
```

Currently, `compute_duration()` only uses interval to calculate "duration", would it be okay to add a feature to choose between interval and duration?

https://github.com/pharmaverse/admiral/blob/6dead5ad9b9fe529d8b459ae718f8f35cb52296f/R/compute_duration.R#L158

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by ddsjoberg**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #1875 into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/compute_duration.R [hunk 1]
```diff
@@ -157,7 +187,13 @@ compute_duration <- function(start_date,
   }
 
   # derive the duration in the output unit
-  duration <- time_length(start_date %--% end_date, unit = out_unit)
+  duration <-
+    switch(type,
```
Comment (bms63) [inline R/compute_duration.R:191]: YES!!! `switch` I love this function!!  So much easier to understand.  @pharmaverse/admiral a good function to keep in your back pocket!

image

File: tests/testthat/test-compute_duration.R
Comment (bms63) [inline tests/testthat/test-compute_duration.R:1]: Do we need to do a test with just the default, i.e. days?   and what about some ymd_hms?  Now my brain is turning....i don't see any partial date issues, which is something we encounter all the time.  Perhaps a separate issue for us to investigate or do you have time to put in some bad dates??

image

**Review by bms63 — COMMENTED** [review]
Sorry for being memelord in this review!!

File: tests/testthat/test-compute_duration.R
Comment (ddsjoberg) [reply tests/testthat/test-compute_duration.R:1]: I think this PR should include tests about durations vs intervals. Perhaps the bad dates should be a separate issue? (Also, I am not sure I 100% follow 😆 )

I did just add a few tests asserting that the duration and interval types are equivalent for types types like days and weeks, where the results are the same.

Comment (bms63) [reply tests/testthat/test-compute_duration.R:1]: I agree separate issue is best partial dates.  I was interested if anything untoward would happen with 

```
compute_duration(
      ymd("2000-02-01"),
      ymd("2001-02---"),
      out_unit = "weeks",
      add_one = FALSE,
      type = "duration"
    )
  )
```

**Comment by bms63** [comment]
@zdz2101 this all looks good to me.  I made a new issue for my concerns #2077.  Can you approve as well and we can get this one in.

**Review by zdz2101 — APPROVED** [review]

**Comment by bms63** [comment]
@ddsjoberg thanks again for this!!

**Pull request merged**